src/components/SamlConfiguration/index.test.jsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+
+import Index from './index';
+import LmsApiService from './../../data/services/LmsApiService';
+import { REQUIRED_DATA_FIELDS } from './SamlProviderDataForm';
+
+const formData = new FormData();
+REQUIRED_DATA_FIELDS.forEach(field => formData.append(field, 'testdata'));
+
+describe('<SamlConfiguration /> ', () => {
+  it('creating provider config states', () => {
+    const wrapper = shallow(<Index enterpriseId="testEnterpriseId" />);
+    const spyPostNewProviderConfig = jest.spyOn(LmsApiService, 'postNewProviderConfig');
+
+    const mockResponseData = { data: {} };
+    formData.forEach((value, key) => {
+      mockResponseData.data[key] = value;
+    });
+
+    spyPostNewProviderConfig.mockImplementation(() => mockResponseData);
+    wrapper.instance().createProviderConfig(formData, () => {
+      expect(wrapper.state().providerConfig).toEqual(mockResponseData.data);
+    });
+  });
+
+  it('updating provider config states', () => {
+    const wrapper = shallow(<Index enterpriseId="testEnterpriseId" />);
+    const spyPostUpdatedProviderConfig = jest.spyOn(LmsApiService, 'updateProviderConfig');
+
+    const oldProviderConfigData = { data: {} };
+    const mockResponseData = { data: {} };
+
+    formData.forEach((value, key) => {
+      oldProviderConfigData.data[key] = 'old '.concat(value);
```

Comment:
If you don't do the top level thing I suggest, at the very least please change this name.
We have ProviderData and ProviderConfig objects so ProviderConfigData is just goofy lol.